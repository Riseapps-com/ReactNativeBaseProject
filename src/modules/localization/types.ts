export type TranslationGetters = {
  [key: string]: () => Record<string, any>;
};
