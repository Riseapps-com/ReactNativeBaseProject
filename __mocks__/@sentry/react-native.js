module.exports = {
  ...jest.requireActual('@sentry/react-native'),

  init: jest.fn(),
  captureMessage: jest.fn(),
  captureException: jest.fn(),
};
