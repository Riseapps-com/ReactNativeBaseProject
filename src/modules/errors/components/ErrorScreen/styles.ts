import { sizes, styleSheetFactory } from '~theme';

export default styleSheetFactory(theme => ({
  errorScreen: {
    flex: 1,
    backgroundColor: theme.background,
    padding: sizes.SPACING_M,
  },

  content: {
    flex: 1,
    justifyContent: 'center',
  },

  title: {
    color: theme.error,
    textAlign: 'center',
    marginBottom: sizes.SPACING_L,
  },
}));
