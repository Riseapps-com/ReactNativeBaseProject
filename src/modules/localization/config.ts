export const LOCALIZATION_FALLBACK_LANGUAGE = 'en';
