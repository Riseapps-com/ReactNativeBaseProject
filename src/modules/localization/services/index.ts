export { default as I18n } from './I18n';
