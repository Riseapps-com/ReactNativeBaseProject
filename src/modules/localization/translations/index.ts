import { TranslationGetters } from '../types';
import de from './de.json';
import en from './en.json';

const translations: TranslationGetters = {
  en: () => en,
  de: () => de,
};

export { translations };
