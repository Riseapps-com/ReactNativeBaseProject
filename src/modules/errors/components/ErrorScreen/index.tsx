import React from 'react';
import { useTranslation } from 'react-i18next';
import { Button, StatusBar, View } from 'react-native';

import { Text } from '~modules/ui';
import { useTheme } from '~theme';

import themedStyles from './styles';

export type ErrorScreenProps = {
  onDismiss: () => void;
};

const ErrorScreen: React.FC<ErrorScreenProps> = props => {
  const { t } = useTranslation();
  const [styles, theme] = useTheme(themedStyles);

  return (
    <View testID="errorScreen" style={styles.errorScreen}>
      <StatusBar barStyle="light-content" />

      <View style={styles.content}>
        <Text fontStyle={'bold'} size={'bigger'} style={styles.title}>
          {t('errorBoundaryTitle')}
        </Text>

        <Text fontStyle="bold">{t('errorBoundaryTitle')}</Text>
        <Text>{t('errorBoundarySubtitle')}</Text>
      </View>

      <Button onPress={props.onDismiss} title={t('dismiss')} color={theme.primary} />
    </View>
  );
};

export default ErrorScreen;
