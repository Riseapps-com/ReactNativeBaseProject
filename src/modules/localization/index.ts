import * as localizationConfig from './config';

export * from './services';

export { localizationConfig };
