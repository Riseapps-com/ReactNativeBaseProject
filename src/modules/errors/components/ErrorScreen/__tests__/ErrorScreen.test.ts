import { fireEvent, RenderAPI } from '@testing-library/react-native';

import { I18n } from '~modules/localization';
import { renderComponent } from '~modules/tests';

import ErrorScreen from '../index';

describe('errors', () => {
  describe('<ErrorScreen />', () => {
    it('should render correctly', () => {
      const errorScreen: RenderAPI = renderComponent(ErrorScreen);

      expect(errorScreen).toMatchSnapshot();
    });

    it('calls provided function on dismiss', () => {
      const onDismiss = jest.fn();
      const errorScreen: RenderAPI = renderComponent(ErrorScreen, { onDismiss });

      fireEvent.press(errorScreen.getByText(I18n.t('errors.dismiss')));

      expect(onDismiss).toBeCalledTimes(1);
    });
  });
});
