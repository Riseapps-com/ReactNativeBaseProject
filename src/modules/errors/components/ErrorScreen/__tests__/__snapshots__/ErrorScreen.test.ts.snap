// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`errors <ErrorScreen /> should render correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#FFFFFF",
      "flex": 1,
      "padding": 16,
    }
  }
  testID="errorScreen"
>
  <View
    style={
      Object {
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#4A4E4D",
          },
          Object {
            "fontFamily": "Quicksand-Bold",
          },
          Object {
            "fontSize": 18,
          },
          Object {
            "color": "#fe4a49",
            "marginBottom": 24,
            "textAlign": "center",
          },
        ]
      }
    >
      errorBoundaryTitle
    </Text>
    <Text
      style={
        Array [
          Object {
            "color": "#4A4E4D",
          },
          Object {
            "fontFamily": "Quicksand-Bold",
          },
          Object {
            "fontSize": 16,
          },
          undefined,
        ]
      }
    >
      errorBoundaryTitle
    </Text>
    <Text
      style={
        Array [
          Object {
            "color": "#4A4E4D",
          },
          Object {
            "fontFamily": "Quicksand-Regular",
          },
          Object {
            "fontSize": 16,
          },
          undefined,
        ]
      }
    >
      errorBoundarySubtitle
    </Text>
  </View>
  <View
    accessibilityRole="button"
    accessibilityState={Object {}}
    accessible={true}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "margin": 8,
              "textAlign": "center",
            },
            Object {
              "color": "#0036A7",
            },
          ]
        }
      >
        dismiss
      </Text>
    </View>
  </View>
</View>
`;
